### PR TITLE
added reverse connection

### PR DESCRIPTION
**Pull request by nightfury394**

### CODE REVIEW

File: project/Teacher.cs [hunk 1]
```diff
@@ -17,21 +17,52 @@ public class Teacher
...
+            private set => _subjects = value;
+        }
+
         public Availability AvailabilityStatus { get; set; }
 
-        public Teacher(List<Subject> subjects, string name)
+        public Teacher(string name)
         {
-            Name = name;
+            Name = name ?? throw new ArgumentException("Name cannot be null or empty.");
```
Comment (hussalen) [inline project/Teacher.cs]: pointless, since name from the parameter is already non-nullable

File: project/Subject.cs [hunk 2]
```diff
@@ -1,67 +1,60 @@
...
+            private set
+            {
+                if (value < 1 || value > 5)
+                    throw new ArgumentException("Grading scale must be between 1 and 5.");
+                _gradingScale = value;
+            }
         }
 
-        if (_subjectList.Any(s => s.SubjectId == subject.SubjectId))
+        private Teacher _teacher;
```
Comment (hussalen) [inline project/Subject.cs]: Teacher <=> Students is 1..* <=> 1..*, meaning you can't have just the field Teacher, but you need the List<Teacher>, just like how you have the List<Subjects> in Teacher

**Review by hussalen — CHANGES_REQUESTED** [review]
Overall much better than last time, please update based on my comments @nightfury394

File: project/Teacher.cs (hunk 1 above)
Comment (nightfury394) [reply project/Teacher.cs]: Okay . Resolved

**Review by hussalen — APPROVED** [review]

**Pull request merged**